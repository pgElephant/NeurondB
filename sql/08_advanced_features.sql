-- Test sync (should not crash)
SELECT sync_index_async('test_index', 'replica_host') AS sync_initiated;

-- Test array conversion roundtrip
SELECT vector_to_array(array_to_vector(ARRAY[1.0, 2.0, 3.0]::real[]));

-- Verify catalog integrity
SELECT COUNT(*) >= 12 AS all_catalogs_present 
FROM pg_tables 
WHERE tablename LIKE 'neurondb_%';

-- Verify extension info
SELECT extname, extversion 
FROM pg_extension 
WHERE extname = 'neurondb';

