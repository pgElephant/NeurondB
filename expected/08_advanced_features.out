-- Test sync (should not crash)
SELECT sync_index_async('test_index', 'replica_host') AS sync_initiated;
ERROR:  function sync_index_async(unknown, unknown) does not exist
LINE 1: SELECT sync_index_async('test_index', 'replica_host') AS syn...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Test array conversion roundtrip
SELECT vector_to_array(array_to_vector(ARRAY[1.0, 2.0, 3.0]::real[]));
ERROR:  function array_to_vector(real[]) does not exist
LINE 1: SELECT vector_to_array(array_to_vector(ARRAY[1.0, 2.0, 3.0]:...
                               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
-- Verify catalog integrity
SELECT COUNT(*) >= 12 AS all_catalogs_present 
FROM pg_tables 
WHERE tablename LIKE 'neurondb_%';
 all_catalogs_present 
----------------------
 f
(1 row)

-- Verify extension info
SELECT extname, extversion 
FROM pg_extension 
WHERE extname = 'neurondb';
 extname | extversion 
---------+------------
(0 rows)

